### ISSUE TITLE
Add inline-rst feature

### ISSUE THREAD

**Issue by santiagopagani** [boschglobal/doxysphinx#73]
At the moment doxysphinx uses `@verbatim embed:rst:leading-asterisk` and `@endverbatim` to input a block of rst code.
It would also be nice to add the inline-rst feature, just like Breathe currently supports: [inline-rst with Breathe](https://breathe.readthedocs.io/en/latest/markups.html#inline-rst).

Our current use case is to have links to terms in the glossary of Sphinx-Needs, which we would like to be able to refer to from the Doxygen documentation of some class or function. I can think of several other use cases as well.

For example, by defining two extra aliases in the Doxyfile:
```
ALIASES               += rst_inline="@verbatim embed:rst:inline"
ALIASES               += endrst_inline="@endverbatim"
```

We could then have some Doxygen comments that look like this (note that we have both normal rst blocks and the inline-rst as well:
```
    /// Using some glossary inline with custom alias:
    /// First glossary term @rst_inline :term:`Term A` @endrst_inline and
    /// another one @rst_inline :term:`Term B` @endrst_inline and more text.
    ///
    /// @rst
    /// .. comp-sw:: Title
    ///    :id: COMP_SW_ID
    ///    :implements: ARCH_SW_ID
    /// @endrst
```

This example works perfectly fine with Breathe. Doxysphinx however simply ignores what is written in the `rst_inline`.

**Comment by twodrops** [comment]
@mb-emag will check this. This should have worked out-of-the-box.

**Comment by mb-emag** [comment]
I looked into the issue - thought it would be easier...

Verbatim elements with `embed:rst:inline` will be parsed and interpreted as rst content by us, however there's a bug that will then swallow this single line and the result is that nothing gets rendered out.

But fixing that bug aside there is another problem: 

Sphinx adds paragraphs (`<p>`-tags) for each block (directives, content etc.).

That means that something like this (which we use to split up the html and insert our rst snippets - in this case this would be the line starting with `:doc:`):
```rst
====================
Raw HTML inline test
====================

This is a test for inline raw HTML

.. raw:: html

   <span style="background: red">This is an inline test... now this was the prefix...

:doc:`This is a restructured text link onto the main index <../index>`

.. raw:: html

   now this is the suffix </span>
```

will result in an html like that (after sphinx ran):
```html
<h1>Raw HTML inline test<a class="headerlink" href="#raw-html-inline-test" title="Permalink to this headline">¶</a></h1>
<p>This is a test for inline raw HTML</p>
<span style="background: red">This is an inline test... now this was the prefix...
<!-- The wrapping paragraph (<p>-tag) on the next line is our problem -->
<p><a class="reference internal" href="../index.html"><span class="doc">This is a restructured text link onto the main index</span></a></p>
now this is the suffix </span></div>
```
This means that even if we would parse the inline rst correctly it cannot be rendered inline correctly. And as doxysphinx is no sphinx extension we cannot change the rst doctree here because we're too early in the process...

**However there is still some implementation possibility**:
I could inject a simple javascript for each inline rst snippet that will remove the wrapping `<p>`-tag and then make it inline. However this would feel like a really dirty hack...

@twodrops what do you think?

**Comment by mb-emag** [comment]
Finally solved in release 3.1.0.
The documentation on the inline-rst bits is here: <https://boschglobal.github.io/doxysphinx/docs/syntax/rst_inline_syntax.html>
